### PR TITLE
First iteration of Smart Links

### PR DESCRIPTION
**Pull request by Jnegrier**

### CODE REVIEW

**Comment by opishcheiko** [comment]
@Jnegrier let me know where exactly you want me to have a look

**Comment by Jnegrier** [comment]
> @Jnegrier let me know where exactly you want me to have a look

Thanks, don't worry, I fixed the piece where I was stuck before, the last commit has a lot of changes

File: components/templates/Test.tsx [hunk 1]
```diff
@@ -45,6 +45,17 @@ const useStyles = makeStyles(theme => ({
...
   textColorLightMode: {
   },
+  externalLogsButton: {
+    backgroundColor: theme.palette.secondary.dark,
+    width: "auto",
+    height: "auto",
+    marginLeft: "10px",
+    border: "1px #696969 solid",
+    fontSize: "12px",
+    color: "#7CFC00",
```
Comment (opishcheiko) [inline components/templates/Test.tsx:55]: Juan, do you mind using "green" for consistency of the colours? 😄 Or something less neon/aggressive

Comment (Jnegrier) [reply components/templates/Test.tsx:55]: hahahaha sure @opishcheiko  😄 you can actually ignore this style. Cause the user will be able to pick the color for this type of buttons 

Screenshot 2020-11-25 at 17 02 46

Comment (opishcheiko) [reply components/templates/Test.tsx:55]: ahaha, whyyyyy are you so cruel 😢 , you want to let user destroy the perfect colour palette of our site 👩‍🎨

File: components/templates/settings/ProjectSettingsTabs.tsx [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        {/* <TabPanel value={value} index={1} dir={theme.direction}>
+          Contents for second menu
+        </TabPanel>
+        <TabPanel value={value} index={2} dir={theme.direction}>
+          Contents for third menu
+        </TabPanel> */}
+      </SwipeableViews>
+    </div>
+  )
+}
```
Comment (opishcheiko) [inline components/templates/settings/ProjectSettingsTabs.tsx:100]: do you think it's beneficial to have it as a separate component?

File: pages/index.tsx [hunk 3]
```diff
@@ -116,6 +117,13 @@ const useStyles = makeStyles(theme => ({
     boxShadow: theme.shadows[5],
     padding: theme.spacing(10, 15, 3),
   },
+  settingsButton: {
```
Comment (opishcheiko) [inline pages/index.tsx:120]: settingsButton not needed?

File: pages/tests/[testsByRunId].tsx [hunk 4]
```diff
@@ -303,21 +311,7 @@ function Tests({
...
-                              "?status=fail"
-                            }
-                            className={classes.spectreButton}
-                            target="_blank"
-                          >
-                            Spectre
-                          </Button>
-                        </div>
-                      ) : null}
+                      <SmartLinksTestRun project_id={test_run.project_id} environment={test_run.environment} test_run_id={test_run.test_run_id}/>
```
Comment (opishcheiko) [inline pages/tests/[testsByRunId].tsx:314]: let's have a quick run through on how to set up a spectre link.
Or better a tooltip for the user to understand this functionality.

**Review by opishcheiko — APPROVED** [review]
looks good to me, thanks Juan! Big feature!

File: components/templates/settings/ProjectSettingsTabs.tsx (hunk 2 above)
Comment (Jnegrier) [reply components/templates/settings/ProjectSettingsTabs.tsx:100]: @opishcheiko Yeah :) Its design in a way that is easier to add more menus with more options for a project, so its perfect for the future

File: components/templates/Test.tsx (hunk 1 above)
Comment (Jnegrier) [reply components/templates/Test.tsx:56]: Please feel free to change this palette of colors to a way to like the most 👍

File: pages/index.tsx (hunk 3 above)
Comment (Jnegrier) [reply pages/index.tsx]: Nope, just removed the style for it, thanks for the heads up

**Pull request merged**